src/index.test.js

```diff
@@ -0,0 +1,122 @@
+{
+    var result = Index.getProjectVersion('<project><version>1.0.0</version></project>', 'xml');
+    expect(result).toBe('1.0.0');
+});
+
+test('testGetProjectVersionWithPackageJsonFile', async() => 
+{
+    var result = Index.getProjectVersion('{"version":"1.0.0"}', 'json');
+    expect(result).toBe('1.0.0');
+});
+
+test('testGetProjectVersionWithTxtFile', async() => 
+{
+    var result = Index.getProjectVersion('1.0.0', 'txt');
+    expect(result).toBe('1.0.0');
+});
+
+test('testGetProjectVersionWithUnsupportedFile', async() => 
+{
+    var result = Index.getProjectVersion('1.0.0', 'md');
+    expect(result).toBe(undefined);
+});
+
+it('testCheckVersionUpdateWithVersionsAreEqual', async() => 
+{
+    // action
+    Index.checkVersionUpdate('1.0.0', '1.0.0', undefined);
+
+    // verify
+    expect(core.setFailed).toHaveBeenCalledWith('You have to update the project version!');
+});
+
+it('testCheckVersionUpdateWithVersionIsDowngraded', async() => 
+{
+    // action
+    Index.checkVersionUpdate('1.0.0', '0.9.0', undefined);
+
+    // verify
+    expect(core.setFailed).toHaveBeenCalledWith('You have to update the project version!');
+});
+
+it('testCheckVersionUpdateWithVersionIsUpdated', async() => 
+{
+    // action
+    Index.checkVersionUpdate('1.0.0', '1.1.0', undefined);
+
+    // verify
+    expect(core.setFailed).not.toHaveBeenCalledWith('You have to update the project version!');
+});
+
+it('testCheckVersionUpdateWithVersionIsUpdatedAndAdditionalFilesGivenButNotUpdated', async() => 
+{
+    // prepare
+    fs.readFileSync.mockReturnValue('foo... version: 1.0.0 ...bar');
+
+    // action
+    Index.checkVersionUpdate('1.0.0', '1.1.0', ['README.md']);
+
+    // verify
+    expect(fs.readFileSync).toHaveBeenCalledWith('test/workspace/README.md');
+    expect(core.setFailed).toHaveBeenCalledWith('You have to update the project version in "README.md"!');
+});
+
+it('testCheckVersionUpdateWithVersionIsUpdatedAndAdditionalFilesGiven', async() => 
+{
+    console.log(process.env.GITHUB_WORKSPACE);
+    // prepare
+    fs.readFileSync.mockReturnValue('foo... version: 1.1.0 ...bar');
+
+    // action
+    Index.checkVersionUpdate('1.0.0', '1.1.0', ['README.md']);
+
+    // verify
+    expect(fs.readFileSync).toHaveBeenCalledWith('test/workspace/README.md');
+    expect(core.setFailed).not.toHaveBeenCalledWith('You have to update the project version in "README.md"!');
+});
+
+it('testCheckVersionUpdateWithVersionIsUpdatedAndAdditionalFilesGivenWithSpaceInString', async() => 
+{
+    console.log(process.env.GITHUB_WORKSPACE);
+    // prepare
+    fs.readFileSync.mockReturnValue('foo... version: 1.1.0 ...bar');
+
+    // action
+    Index.checkVersionUpdate('1.0.0', '1.1.0', [' README.md']);
+
+    // verify
+    expect(fs.readFileSync).toHaveBeenCalledWith('test/workspace/README.md');
+    expect(core.setFailed).not.toHaveBeenCalledWith('You have to update the project version in "README.md"!');
+});
+
```

Comment:
Kann raus

Reply:
Nein. Hier wird ein anderer Fall getestet, bei dem in dem Array der `additional_files_to_check` ein Leerzeichen im Dateinamen enthalten ist, da beim Aufbau der Action die Komma separierte Liste bspw. `datei1.xml, datei2.xml` aufgebaut wurde.